### PR TITLE
fix: propagate config build error instead of panicking

### PR DESCRIPTION
**Pull request by pront**
## Motivation
fixes: https://github.com/vectordotdev/vector/issues/18110

## Summary
Propagate an error up the stack instead of panicking at deserializer instantiation.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Thanks!

File: src/components/validation/runner/mod.rs [hunk 1]
```diff
@@ -413,7 +413,7 @@ fn build_external_resource(
     configuration: &ValidationConfiguration,
     input_task_coordinator: &TaskCoordinator<Configuring>,
     output_task_coordinator: &TaskCoordinator<Configuring>,
-) -> (RunnerInput, RunnerOutput, Option<Encoder<encoding::Framer>>) {
+) -> Result<(RunnerInput, RunnerOutput, Option<Encoder<encoding::Framer>>), String> {
```
Comment (bruceg) [inline src/components/validation/runner/mod.rs:416]: Is there a more specific error type that could be returned here?

**Review by bruceg — DISMISSED** [review]
More changes than I expected, but nothing particularly surprising. LGTM

**Comment by pront** [comment]
Made a small commit to change the error type so this PR now need new approvals.

**Review by bruceg — APPROVED** [review]

**Pull request merged**